### ISSUE TITLE
Titanet Small For Language Identification not training properly . . . .

### ISSUE THREAD

**Issue by programindz** [NVIDIA-NeMo/NeMo#14504]
I'm trying to train a small titanet model for lid over 8k audio samples for 18 languages. I have checked every step and from preparing the manifest jsonl files to feature extraction, everything seems perfect but the model is not training at all.

<img width="380" height="409" alt="Image" src="https://github.com/user-attachments/assets/b406ef3f-3418-4d2b-b75d-1b56c9ce5a22" />

check this training loss, which i'm unable understand why it is happening. I have even changed the batch sizes and learning rates but nothing has helped me so for.

I have around 1k hours of data for training (for all languages). 

My current titanet_small.yaml config:
--
name: &name "TitaNet-S"
sample_rate: &sample_rate 8000

model:
  train_ds:
    manifest_filepath: train.jsonl
    sample_rate: 8000
    labels: null
    batch_size: 16
    shuffle: True
    is_tarred: False
    tarred_audio_filepaths: null
    tarred_shard_strategy: "scatter"
    augmentor:
      noise:
        manifest_path: noise.jsonl
        prob: 0.3
        min_snr_db: 5
        max_snr_db: 10

      impulse:
        manifest_path: rir.jsonl
        prob: 0.3 

      speed:
        prob: 0.3
        sr: *sample_rate
        resample_type: 'kaiser_fast'
        min_speed_rate: 0.90
        max_speed_rate: 1.10

  validation_ds:
    manifest_filepath: valid.jsonl
    sample_rate: 8000
    labels: null
    batch_size: 16
    shuffle: False

  model_defaults:
    filters: 256
    repeat: 3
    dropout: 0.1
    separable: true
    se: true
    se_context_size: -1
    kernel_size_factor: 1.0
    
  preprocessor:
    _target_: nemo.collections.asr.modules.AudioToMelSpectrogramPreprocessor
    normalize: "per_feature"
    window_size: 0.025
    sample_rate: *sample_rate
    window_stride: 0.01
    window: "hann"
    features: &n_mels 60
    n_fft: 512
    frame_splicing: 1
    dither: 0.00001

  spec_augment:
    _target_: nemo.collections.asr.modules.SpectrogramAugmentation
    freq_masks: 2
    freq_width: 2
    time_masks: 3
    time_width: 0.03

  encoder:
    _target_: nemo.collections.asr.modules.ConvASREncoder
    feat_in: *n_mels
    activation: relu
    conv_mask: true

    jasper:
      -   filters: ${model.model_defaults.filters}
          repeat: 1
          kernel: [3]
          stride: [1]
          dilation: [1]
          dropout: 0.0
          residual: false
          separable: ${model.model_defaults.separable}
          se: ${model.model_defaults.se}
          se_context_size: ${model.model_defaults.se_context_size}

      -   filters: ${model.model_defaults.filters}
          repeat:  ${model.model_defaults.repeat}
          kernel: [7]
          stride: [1]
          dilation: [1]
          dropout: ${model.model_defaults.dropout}
          residual: true
          separable: ${model.model_defaults.separable}
          se: ${model.model_defaults.se}
          se_context_size: ${model.model_defaults.se_context_size}

      -   filters: ${model.model_defaults.filters}
          repeat: ${model.model_defaults.repeat}
          kernel: [11]
          stride: [1]
          dilation: [1]
          dropout: ${model.model_defaults.dropout}
          residual: true
          separable: ${model.model_defaults.separable}
          se: ${model.model_defaults.se}
          se_context_size: ${model.model_defaults.se_context_size}

      -   filters: ${model.model_defaults.filters}
          repeat: ${model.model_defaults.repeat}
          kernel: [15]
          stride: [1]
          dilation: [1]
          dropout: ${model.model_defaults.dropout}
          residual: true
          separable: ${model.model_defaults.separable}
          se: ${model.model_defaults.se}
          se_context_size: ${model.model_defaults.se_context_size}

      -   filters: &enc_feat_out 3072
          repeat: 1
          kernel: [1]
          stride: [1]
          dilation: [1]
          dropout: 0.0
          residual: false
          separable: ${model.model_defaults.separable}
          se: ${model.model_defaults.se}
          se_context_size: ${model.model_defaults.se_context_size}

  decoder:
    _target_: nemo.collections.asr.modules.SpeakerDecoder
    feat_in: *enc_feat_out
    num_classes: 18 # modified
    pool_mode: 'attention'
    emb_sizes: 192

  loss:
    _target_: nemo.collections.asr.losses.angularloss.AngularSoftmaxLoss
    scale: 30
    margin: 0.2

  optim:
    name: sgd
    lr: .006 #(original titanet-large was trained with 0.08 lr)
    weight_decay: 0.001
    momentum: 0.9

    # scheduler setup
    sched:
      name: CosineAnnealing
      warmup_ratio: 0.1
      min_lr: 0.0

trainer:
  devices: 3 # number of gpus (original titanet-large was trained on 4 nodes with 8 gpus each)
  max_epochs: 20
  max_steps: -1 # computed at runtime if not set
  num_nodes: 1
  accelerator: gpu
  strategy: ddp
  accumulate_grad_batches: 1
  deterministic: False
  enable_checkpointing: False
  logger: False
  log_every_n_steps: 1  # Interval of logging.
  val_check_interval: 1.0  # Set to 0.25 to check 4 times per epoch, or an int for number of iterations
  gradient_clip_val: 1.0

exp_manager:
  exp_dir: ./titanet_lid
  name: *name
  create_tensorboard_logger: True
  create_checkpoint_callback: True

**Comment by nithinraok** [comment]
Sorry for delayed response, missed this issue. 

marble net is trained with titanet architecture, have you checked this config: https://github.com/NVIDIA-NeMo/NeMo/blob/main/examples/asr/conf/lang_id/titanet_large.yaml

**Comment by nithinraok** [comment]
As there has been no response, I am closing this as completed. Please feel free to reopen it if the suggested solution does not work.
